File: CS/TileLayoutControlDescendant/TileLayoutCntrlDescendant.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.Xpf.LayoutControl;
using DevExpress.Xpf.Core;
using System.Windows;
using DevExpress.Xpf.Core.Native;

namespace TileLayoutControlDescendant
{
    public class TileLayoutCntrlDescendant : TileLayoutControl 
    {
        public TileLayoutCntrlDescendant():base() { }
        protected override PanelControllerBase CreateController()
        {
            return new TileLayoutDescendantController(this);
        }
    }

    public class TileLayoutDescendantItemDragAndDropController : TileLayoutItemDragAndDropController
    {
        public TileLayoutDescendantItemDragAndDropController(Controller controller, Point startDragPoint, FrameworkElement dragControl): base(controller, startDragPoint, dragControl)
        {
            
        }

        public override void DragAndDrop(Point p)
        {
            FrameworkElement ctrl = this.Controller.Control;
            if (!LayoutHelper.IsPointInsideElementBounds(p, ctrl, new Thickness()))
            {
                DragControl.SetVisible(false);
                DataObject dragData = new DataObject("DragDropDataDG", DragControl.DataContext);
                DragDrop.DoDragDrop(ctrl, dragData, DragDropEffects.Copy);
                DragControl.SetVisible(true);
                return;
            } 
            base.DragAndDrop(p);
        }
    }

    public class TileLayoutDescendantController: TileLayoutController
    {
        public TileLayoutDescendantController(IFlowLayoutControl control) : base(control) { }

        protected override DragAndDropController CreateItemDragAndDropControler(Point startDragPoint, FrameworkElement dragControl)
        {
            return new TileLayoutDescendantItemDragAndDropController(this, startDragPoint, dragControl);
        }
    }
}

File: VB/TileLayoutControlDescendant/TileLayoutCntrlDescendant.vb
Imports DevExpress.Xpf.LayoutControl
Imports DevExpress.Xpf.Core
Imports System.Windows
Imports DevExpress.Xpf.Core.Native

Namespace TileLayoutControlDescendant

    Public Class TileLayoutCntrlDescendant
        Inherits TileLayoutControl

        Public Sub New()
            MyBase.New()
        End Sub

        Protected Overrides Function CreateController() As PanelControllerBase
            Return New TileLayoutDescendantController(Me)
        End Function
    End Class

    Public Class TileLayoutDescendantItemDragAndDropController
        Inherits TileLayoutItemDragAndDropController

        Public Sub New(ByVal controller As Controller, ByVal startDragPoint As Point, ByVal dragControl As FrameworkElement)
            MyBase.New(controller, startDragPoint, dragControl)
        End Sub

        Public Overrides Sub DragAndDrop(ByVal p As Point)
            Dim ctrl As FrameworkElement = Controller.Control
            If Not LayoutHelper.IsPointInsideElementBounds(p, ctrl, New Thickness()) Then
                DragControl.SetVisible(False)
                Dim dragData As DataObject = New DataObject("DragDropDataDG", DragControl.DataContext)
                DragDrop.DoDragDrop(ctrl, dragData, DragDropEffects.Copy)
                DragControl.SetVisible(True)
                Return
            End If

            MyBase.DragAndDrop(p)
        End Sub
    End Class

    Public Class TileLayoutDescendantController
        Inherits TileLayoutController

        Public Sub New(ByVal control As IFlowLayoutControl)
            MyBase.New(control)
        End Sub

        Protected Overrides Function CreateItemDragAndDropControler(ByVal startDragPoint As Point, ByVal dragControl As FrameworkElement) As DragAndDropController
            Return New TileLayoutDescendantItemDragAndDropController(Me, startDragPoint, dragControl)
        End Function
    End Class
End Namespace
